doc/userguide/output/eve/eve-json-output.rst

```diff
@@ -301,6 +307,32 @@ This example will cause each Suricata thread to write to its own "eve.json" file
 by adding a unique identifier to the filename.  For example, ``eve.7.json``.
 
 
+.. _output_eve_flush:
+
+Flushing file contents
+--~~~~~~~~~~~~~~~~~~~~
+
+By default, Suricata will call ``flush`` each time the file is written. ``flush-threshold``
+can be set to a byte-count to reduce the number of ``fflush`` calls. When non-zero, ``fflush`` is called
+when the bytes written to the file are equal to or exceed the value; subsequent calls to ``fflush`` occur.
+
+Reducing the number of ``fflush`` calls can reduce write overhead at the expense of risk of leaving
+up to ``flush-threshold`` bytes unbuffered/unwritten should Suricata unexpectedly terminate.
+
+The default value is ``0``.
+
+::
+
+   outputs:
+     - eve-log:
+         filename: eve.json
+         flush-threshold: 1024
```

Comment:
As no alternatives to the default are provided, users may look at this example as a suggestion, however, given average log sizes a value of 1024 might not help much - might reduce the flush to every 2nd log record. Would 4k or 8k be more reasonable?

Reply:
I'll update the example.

However, the timing issue (lingering data) is a problem that I have to solve if this PR is to move forward.